### ISSUE TITLE
change hard dependency on @vue/compiler-sfc

### ISSUE THREAD

**Issue by stefpb** [vuejs/vue-jest#469]
On 
https://github.com/vuejs/core/tree/main/packages/compiler-sfc i see this note:

> Note: as of 3.2.13+, this package is included as a dependency of the main vue package and can be accessed as vue/compiler-sfc. This means you no longer need to explicitly install this package and ensure its version match that of vue's. Just use the main vue/compiler-sfc deep import instead.

but after i removed this package i get this error when i run jest:

**Error: @vue/vue3-jest tried to access @vue/compiler-sfc, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.** 

Should we change this behavior or should i create a PR for https://github.com/vuejs/core/tree/main/packages/compiler-sfc ?

**Comment by lmiller1990** [comment]
We should support what Vue core recommends. I'm wondering where that error is even coming from - I didn't know Jest did dependency checks.

What version of Jest is this? In general, I think it's good to explicitly list all your dependencies, even if it's not strictly required.

**Comment by lmiller1990** [comment]
Is this error originating from Jest? What package manager are you using? It seems VERY weird that Jest knows or cares about what versions you are using.

**Comment by stefpb** [comment]
Yes, the error comes when i execute `yarn test` or `yarn jest`. I use yarn berry in version 3.2.1.

My proposed change is to follow the note on https://github.com/vuejs/core/tree/main/packages/compiler-sfc : you don't need to use it, because it is build in: 

> Note: as of 3.2.13+, this package is included as a dependency of the main vue package and can be accessed as vue/compiler-sfc. This means you no longer need to explicitly install this package and ensure its version match that of vue's. Just use the main vue/compiler-sfc deep import instead.

You don't have the package @vue/compiler-sfc in your package.json?

**Comment by stefpb** [comment]
So i did now `yarn remove @vue/compiler-sfc` and `yarn add -P @vue/compiler-sfc`. But after run `yarn test` i get again
```
@vue/vue3-jest tried to access @vue/compiler-sfc, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

    Required package: @vue/compiler-sfc
    Required by: @vue/vue3-jest@virtual:ced3ea7f25b053dfea2569066b422024c114955218d4f499fdb5b4f8313182d76ffb265b92d7e5a0a8e0bfc12c58e2f40968c70254065c492df6caac66093f79#npm:27.0.0 (via /var/www/html/assets/.yarn/__virtual__/@vue-vue3-jest-virtual-bfe7435b5c/0/cache/@vue-vue3-jest-npm-27.0.0-e65b085f40-6bec574d3b.zip/node_modules/@vue/vue3-jest/lib/)
```

![image](https://user-images.githubusercontent.com/1069441/174585109-fe399489-2a98-48db-a318-5dd6b283b939.png)
